node-cpp.md

```diff
@@ -0,0 +1,128 @@
+# Node.js C++ Addons
+
+The following document outlines Mapbox's approach to writing native abstractions for Node.js (anecdotally referred to as _addons_).
```

Comment:
`native abstractions for Node.js` is a description of [Nan](https://github.com/nodejs/nan) not node addons, so I would change to `The following document outlines Mapbox's approach to writing C++ modules for Node.js (often referred to as _addons_)`

 - A node C++ addon is a module, writting in C++ that extends nodejs with functionality
 - Nan is a header that makes it easier to write a c++ addon.
 - You can write a C++ addon without Nan.